### PR TITLE
use `as_ptr` to determine the address of atomics

### PR DESCRIPTION
**Pull request by stlankes**
The PR #107736 renamed  atomic `as_mut_ptr` to `as_ptr`. Consequently, the futex implementation of the tier-3 platform `RutyHermit` has to use this new interface. In addition, this PR removes also an unused import.

### CODE REVIEW

File: library/std/src/sys/hermit/futex.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ pub fn futex_wait(futex: &AtomicU32, expected: u32, timeout: Option<Duration>) -
 
     let r = unsafe {
         abi::futex_wait(
-            futex.as_mut_ptr(),
+            futex.as_ptr() as *mut _,
```
Comment (joboet) [inline library/std/src/sys/hermit/futex.rs:19]: The returned pointer is still mutable, so this does not need casting.
```suggestion
            futex.as_ptr(),
```

Comment (stlankes) [reply library/std/src/sys/hermit/futex.rs]: You are right. I fixed it.

**Comment by stlankes** [comment]
I think that m-ou-se doesn't have the time.

r?

**Comment by stlankes** [comment]
@m-ou-se Thanks!

**Pull request merged**